cpu/mips_pic32_common/include/periph_cpu_common.h

```diff
@@ -51,6 +53,42 @@ enum {
  */
 #define PERIPH_TIMER_PROVIDES_SET
 
+/**
+ * @brief   Use some common SPI functions
+ * @{
+ */
+#define PERIPH_SPI_NEEDS_INIT_CS
+#define PERIPH_SPI_NEEDS_TRANSFER_BYTE
+#define PERIPH_SPI_NEEDS_TRANSFER_REG
+#define PERIPH_SPI_NEEDS_TRANSFER_REGS
+/** @} */
+
+/**
+ * @brief   Override SPI clock speed values
+ * @{
+ */
+#define HAVE_SPI_CLK_T
+typedef enum {
+    SPI_CLK_100KHZ =   100000U, /**< drive the SPI bus with 100KHz */
+    SPI_CLK_400KHZ =   400000U, /**< drive the SPI bus with 400KHz */
+    SPI_CLK_1MHZ   =  1000000U, /**< drive the SPI bus with 1MHz */
+    SPI_CLK_5MHZ   =  5000000U, /**< drive the SPI bus with 5MHz */
+    SPI_CLK_10MHZ  = 10000000U  /**< drive the SPI bus with 10MHz */
+} spi_clk_t;
+/** @} */
+
+/**
+ * @brief   SPI device configuration
+ */
+typedef struct {
+    volatile uint32_t *mosi_reg;    /**< Output pin mux register address */
+    volatile uint32_t *miso_reg;    /**< MISO pin mux register address */
+    uint8_t mosi_af;                /**< Specify function of output pin */
+    uint8_t miso_af;                /**< Specify input pin for MISO */
+    gpio_t mosi_pin;                /**< GPIO pin for MOSI */
```

Comment:
I couldn't get this code to compile as is, as there is a circular dependency starting with gpio_t here (gpio.h eventually needs spi_conf_t, as a quick hack I just declared gpio_t here:

```
--- a/cpu/mips_pic32_common/include/periph_cpu_common.h
+++ b/cpu/mips_pic32_common/include/periph_cpu_common.h
@@ -80,6 +80,8 @@ typedef enum {
 /**
  * @brief   SPI device configuration
  */
+typedef unsigned int gpio_t;
+

```